ROADMAP.md

```diff
@@ -0,0 +1,93 @@
+Roadmap
+=======
+
+The purpose of this repo is to offer ways to read and write UnixFSv1 structures using standard (go-ipld-prime) `ipld.Node` and `ipld.NodeAssembler`,
+and also to "path" over these structures in natural ways.
+
+This will be done by producing several [ADL](https://github.com/ipld/docs/blob/master/docs/advanced-layouts.md)s.
+
+- ADL to make files appear as `kind`==`bytes`, where interally layout=trickledag:
+	- [ ] Readable
+	- [ ] Writable
+		- [ ] Chunker as a callback/interface
+	- (may choose to implement this as mostly shared code with the other file ADL, where the internal layout is just a parameter; undecided)
+	- [ ] Upstream work to determine how `ipld.Node` and `ipld.NodeAssembler` should work for data in the size scale that requires streaming and seeking
+- ADL to make files appear as `kind`==`bytes`, where interally layout=balanceddag:
```

Comment:
> where interally layout=trickledag
> where interally layout=balanceddag

Hm... the wording confuses me: Is the above only applicable to the writer ( for obvious reasons ), or does this apply to the reader as well? That is - can an ADL be written which reads arbitrary arrangements of UnixFS nodes, as originally intended way way back?

Reply:
Probably the writers are different and the readers end up being the same.  But I didn't want to be prescriptive about how it's implemented.  Easy enough to check two boxes at once if the implementer ends up nailing both goals with the same code.

Reply:
Postel violation though. I want go-ipfs to continue being able reading dags created by something that's not go-ipfs.

Reply:
I don't understand what in this wording gives the impression that that will somehow get harder as a result of this work, but if there is such a miswording, I'm open to fixing it.